File: public/js/game/commands/removeRoomUserCommand.js
function RemoveRoomUserCommand(roomId){
    Command.call(this, RemoveRoomUserCommand.ID);
    this.roomId = roomId
    this.data = true;
    this.execute = function(){
        var user = GameUser.instantiate(this.data.user);
        if(Model.room.containsUser(user)){
            user = Model.room.removeUser(user);
            Model.dispatch(Events.ROOM_USERS_CHANGE, user)
            Model.dispatch(Events.ROOM_USER_REMOVE, user)
            trace("User uid=" + user.uid + " removed from room #" + Model.room.roomId);
        }else{
            throw "User " + user.uid + " not found in room " + Model.room.roomId;
        }
    }
}
RemoveRoomUserCommand.ID = 'removeuser'
Command.add(RemoveRoomUserCommand.ID, RemoveRoomUserCommand)

File: src/main/java/com/hellsepontus/commands/RemoveRoomUserCommand.java
package com.hellsepontus.commands;

import com.hellsepontus.model.GameUser;
import com.hellsepontus.model.Room;

import java.util.Collections;

public class RemoveRoomUserCommand extends Command{
    
    public static final String ID = "removeuser";
    
    public GameUser user;

    public RemoveRoomUserCommand(GameUser user) {
        super(ID);
        this.user = user;
    }

    @Override
    public void execute() {
        Room room = user.getRoom();
        room.removeUser(user);
        sendToAll(room, ID, Collections.singletonMap("user", (Object) user));
    }

    @Override
    public GameUser getUser() {
        return user;
    }
}
